bitcoin/src/blockdata/fee_rate.rs

```diff
@@ -79,6 +79,14 @@ impl FeeRate {
     ///
     /// Computes `self / rhs` returning `None` if `rhs == 0`.
     pub fn checked_div(self, rhs: u64) -> Option<Self> { self.0.checked_div(rhs).map(Self) }
+
+    /// Checked weight multiplication.
+    ///
+    /// Computes `self * rhs` where rhs is of type Weight.  `None` is returned if an overflow
+    /// occured.
+    pub fn checked_weight_mul(self, rhs: Weight) -> Option<Amount> {
```

Comment:
Maybe `checekd_mul_by_weight` would sound better?

Reply:
+1

Reply:
Yeah, I thought about it but I couldn't quickly find a name for that and thought being consistent is nice. If you have some name you like feel free to write it. :)

Reply:
I'm fine with `checekd_mul_by_weight`.

@tcharding I was thinking this is simply the checked version of the this [trait](https://github.com/rust-bitcoin/rust-bitcoin/blob/master/bitcoin/src/blockdata/fee_rate.rs#L108) which also returns `Amount`.  If you have an idea of where to place a helper function called `fee()` or something like that which does the same thing that works also.

Reply:
I think just use `checked_mul_by_weight` and leave any further investigation of the API to https://github.com/rust-bitcoin/rust-bitcoin/issues/1639

Reply:
Works for me